### PR TITLE
Add count command

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: libvast/src/system/count_command.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  auto first = invocation.arguments.begin();
+  auto last = invocation.arguments.end();
+  const auto& options = invocation.options;
+  // Read query from input file, STDIN or CLI arguments.
+  std::string query;
+  auto assign_query = [&](std::istream& in) {
+    query.assign(std::istreambuf_iterator<char>{in},
+                 std::istreambuf_iterator<char>{});
+  };
+  if (auto fname = caf::get_if<std::string>(&options, "count.read")) {
```
Comment (mavam) [inline libvast/src/system/count_command.cpp]: This whole dispatching logic seems very similar to `detail::make_(input|output)_stream`. I wonder if we can reuse this logic here.

File: libvast/src/system/application.cpp [hunk 2]
```diff
@@ -64,6 +65,15 @@ auto make_root_command(std::string_view path) {
                                       "definitions"));
 }
 
+auto make_count_command() {
+  return std::make_unique<command>(
+    "count", "count hits for a query without exporting data", "",
+    opts("?count")
+      .add<bool>("node,N", "spawn a node instead of connecting to one")
```
Comment (mavam) [inline libvast/src/system/application.cpp]: The whole `-N` thing seems to be an orthogonal concern. I wonder if it can somehow be factored. @tobim, any ideas?

File: libvast/src/system/count_command.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/count_command.cpp]: I'm not quite sure I grok `make_input_stream` after looking at it for a bit. Do you mean something along the lines of:

```c++
if (auto is = detail::make_input_stream<detail::count>(options))
  assign_query(*is);
```

File: libvast/src/system/application.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/system/application.cpp]: As a side note—currently, the customization point is not executed for `count -N`. We decided to move the customization point out of the start command some time in the future.

File: libvast/src/system/count_command.cpp (hunk 1 above)
Comment (tobim) [reply libvast/src/system/count_command.cpp]: This code does already exist in `sink_command.cpp:52-90`, `make_input_stream` is something different.

File: libvast/src/system/application.cpp (hunk 2 above)
Comment (tobim) [reply libvast/src/system/application.cpp]: We could move the `-N;--node` option to the root command, and check for it in `spawn_or_connect_to_node()`. That would simplify the commands and that function.

Comment (dominiklohmann) [reply libvast/src/system/application.cpp]: I'll get this done in another PR once this is merged.

File: libvast/src/system/count_command.cpp (hunk 1 above)
Comment (tobim) [reply libvast/src/system/count_command.cpp]: Before we create a separate task and the whole ceremony for this, lets just consolidate the functionality within this PR. To make the request more explicit: Please factor the shared logic between `count_command` and `sink_command` into a dedicated function that will be used by both.

**Comment by dominiklohmann** [comment]
[story/ch8984]

File: libvast/src/system/count_command.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/count_command.cpp]: I implemented this as requested and factored it into a shared utility function named `vast::system::read_query` (for the lack of a better name).
